### ISSUE TITLE
CORS policy problem

### ISSUE THREAD

**Issue by WisleyWang** [gradio-app/gradio#3942]
### Describe the bug

hi~  this problem occurs when I visit the page from another machine on the LAN after I have Lunch service locally.

```
Access to font at 'hottps://fonts.gstatic.com/sourcesanspro/v21/6xK3dSBYKcSV-LCoeQqfx1RYOo3qOK71.woff2' fron origin 'https://10.30.21.151:7013' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource
```
my example just like:
···
import gradio as gr

def greet(name):
    return "Hello " + name + "!"

demo = gr.Interface(fn=greet, inputs="text", outputs="text")
    
if __name__ == "__main__":
    demo.launch()   
···

I want to allow CORS, like FLASK or FASTAPI can to set it  allow_origins = ["*"] , but in gradio I never see any metods

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

https://github.com/gradio-app/gradio

### Logs

```shell
*
```

### System Info

```shell
3.27.0
```

### Severity

blocking all usage of gradio

**Comment by abidlabs** [comment]
Hi @WisleyWang can you mount the Gradio app within a larger fastapi application and enable CORS on that fastapi application?

Here's how to do that: https://gradio.app/sharing-your-app/#mounting-within-another-fastapi-app

**Comment by abidlabs** [comment]
Closing for lack of followup

**Comment by Banbury** [comment]
I'm running into the same problem. And no I can't wrap the Gradio app with FastApi, since it isn't my project. I just want to use it.
And as the upstream project, I think it's your responsibility to send the proper headers.

**Comment by abidlabs** [comment]
Hi @Banbury this issue didn't have enough info for us to be able to reproduce and address the issue. Can you please create a new issue with step-by-step instructions to reproduce the problem? Thanks

**Comment by emersonium** [comment]
Hi same issue here, it is a simple problem to describe:

 "Access-Control-Allow-Origin" is part of the HTTP protocol, how do we configure this in gradio ? 


(fastapi has this, so since gradio itself launches fast api, there should be a method to allow setting of this and all HTTP headers.)

thanks!

**Comment by ignaceHelsen** [comment]
Experiencing the exact same problem.
